Automattic/wp-calypso
# Move teams state out of reader

Teams state, which is currently a part of Reader state, is now used in layout, after #48875.

Since Reader state is so large, and since teams state is entirely independent of the rest of Reader, it makes sense for both performance and organisational reasons to extract it into its own top-level piece of state.

This should effectively revert the 7KB (gzipped) increase in `entry-main` size that #48875 introduced.

#### Changes proposed in this Pull Request

* Move `teams` state out of `reader`

#### Testing instructions

* Install [Redux DevTools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en).
* Open DevTools and switch to the Redux tab.
* Open `/home` on the live branch or your local copy of this branch.
* Click `State` on the top right to ensure you see all state, and not just the diff for the last action: 
![image](https://user-images.githubusercontent.com/409615/73774406-d3e87d80-477b-11ea-9f59-3e42cc00101f.png)
* Analyse the tree and verify that there's no `reader` key, even if you expand the list of keys. This indicates that the reader state hasn't been loaded.
* Analyse the tree and verify that there is a `teams` key. This indicates that the teams state has been loaded.
* Verify that navigation displays and works correctly, and that the correct navigation is chosen based on the type of user
